pubtools/pulplib/_impl/model/repository/file.py

```diff
@@ -20,3 +29,66 @@ class FileRepository(Repository):
     )
 
     mutable_urls = attr.ib(default=attr.Factory(lambda: ["PULP_MANIFEST"]), type=list)
+
+    def upload_file(self, file_obj, relative_url=None):
+        """Upload a file to this repository.
+
+        Args:
+            file_obj (str, file object)
+                If it's a string, then it's the path of file to upload.
+                Else, it ought to be a file-like object
+
+            relative_url (str)
+                Path that should be used in remote repository
+
+                If omitted, filename will be used.
```

Comment:
There's no "filename" argument now.  I would rather word it as "the local name of the file will be used".

Also, I guess we'll have to crash if the user passed a file-like object without "name" attribute and without relative_url.  That's probably worth mentioning as well, like:

If `file_obj` is a file object without a `name` attribute, passing `relative_url` is mandatory.